### PR TITLE
feat: Update password page design

### PR DESCRIPTION
**Pull request by bistaastha**
## Clickup
https://app.clickup.com/t/86cx2t8cm

## UI Preview
Screenshot 2024-11-26 at 7 23 49 PM

## Summary by CodeRabbit

- **New Features**
	- Introduced a new "Confirm Password" field in the user registration form.
	- Added a back navigation icon linked to the sign-in page.
	- Implemented a password validation tooltip for enhanced user feedback.
	- Added a `PasswordCheckTooltipComponent` to the shared module.

- **Bug Fixes**
	- Improved form validation indicators for "Full Name" and "Password" fields.

- **Style**
	- Updated layout and styling for the invited user page, enhancing usability and clarity.

### CODE REVIEW

File: src/app/post-verification/invited-user/invited-user.page.ts [hunk 1]
```diff
@@ -51,54 +59,34 @@ export class InvitedUserPage implements OnInit {
...
 
     this.eou$.subscribe((eou) => {
       this.fg.controls.fullName.setValue(eou?.us?.full_name);
+      this.orgName = eou.ou.org_name;
     });
   }
 
-  async saveData() {
+  onPasswordValid(isValid: boolean): void {
+    this.isPasswordValid = isValid;
```
Comment (arjunaj5) [inline src/app/post-verification/invited-user/invited-user.page.ts:82]: please check the case when the password first became valid, but later was changed to an invalid string. @bistaastha

**Review by arjunaj5 — APPROVED** [review]

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/post-verification/invited-user/invited-user.page.html [hunk 2]
```diff
@@ -7,134 +1,116 @@
...
-            *ngIf="!(specialCharValidationDisplay$|async)"
-            class="invited-user--validation__incorrect"
-            name="close-outline"
-          ></ion-icon>
-        </span>
-        <span> Contains atleast 1 special character </span>
+        <form [formGroup]="fg">
+          <div
+            class="invited-user__text invited-user__text-block"
+            [ngClass]="{'invited-user__text__invalid': fg.controls.fullName.touched && !fg.controls.fullName.valid}"
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/post-verification/invited-user/invited-user.page.html:14]: fixed in follow up

File: src/app/post-verification/invited-user/invited-user.page.ts [hunk 3]
```diff
@@ -129,4 +117,21 @@
...
+
+  private passwordMatchValidator(): ValidatorFn {
+    return (): ValidationErrors | null => {
+      const password = this.fg?.controls.password.value;
+      const confirmPassword = this.fg?.controls.confirmPassword.value;
+
+      // Check if passwords match
+      if (password && confirmPassword && password !== confirmPassword) {
+        return { passwordsMismatch: true };
+      }
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/post-verification/invited-user/invited-user.page.ts:133]: fixed in follow up

File: src/app/post-verification/invited-user/invited-user.page.ts [hunk 4]
```diff
@@ -51,54 +59,34 @@
...
-          Validators.minLength(12),
-          Validators.maxLength(32),
-          Validators.pattern(/[A-Z]/),
-          Validators.pattern(/[a-z]/),
-          Validators.pattern(/[0-9]/),
-          Validators.pattern(/[!@#$%^&*()+\-:;<=>{}|~?]/),
-        ]),
-      ],
+      password: ['', [Validators.required, this.customPasswordValidator()]],
+      confirmPassword: ['', [Validators.required, this.passwordMatchValidator()]],
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/post-verification/invited-user/invited-user.page.ts:70]: length is already checked in the password checks tooltip

File: src/app/post-verification/invited-user/invited-user.page.scss [hunk 5]
```diff
@@ -110,3 +148,9 @@
     padding: 0 8px;
   }
 }
+
+.back-icon {
+  margin-bottom: 20px;
+  width: 28px;
+  height: 28px;
+}
```
[the remark this answers is not in the document]
Comment (bistaastha) [reply src/app/post-verification/invited-user/invited-user.page.scss:156]: Not needed

**Pull request closed without merging**